othdrplot/high_density_region_algorithm.py

```diff
@@ -106,69 +110,127 @@ def _inliers_outliers(self, inliers=True):
 
         return cloud
 
-    def drawInliers(self):
+    def drawInliers(self, sample=None):
         """Draw inliers.
 
+        :param sample: Sample of size (n_samples, n_dims).
+        :type sample: :class:`openturns.Sample`
         :return: OpenTURNS Cloud or Pair object if :attr:`self.dim` > 2.
         :rtype: :class:`openturns.Cloud` or :class:`openturns.Pairs`
         """
         return self._inliers_outliers(inliers=True)
 
-    def drawOutliers(self):
+    def drawOutliers(self, sample=None):
         """Draw outliers.
 
+        :param sample: Sample of size (n_samples, n_dims).
+        :type sample: :class:`openturns.Sample`
         :return: OpenTURNS Cloud or Pair object if :attr:`self.dim` > 2.
         :rtype: :class:`openturns.Cloud` or :class:`openturns.Pairs`
         """
         return self._inliers_outliers(inliers=False)
 
-    def drawContour(self):
+    def drawContour(self, drawData=False, drawOutliers=True):
         """Draw contour.
 
-        :return: OpenTURNS Contour object.
-        :rtype: :class:`openturns.Contour`
-        """
-        if self.dim == 2:
-            # Use a regular grid to compute probability response surface
-            X1min = self.sample[:, 0].getMin()[0]
-            X1max = self.sample[:, 0].getMax()[0]
-            X2min = self.sample[:, 1].getMin()[0]
-            X2max = self.sample[:, 1].getMax()[0]
-
-            xx = ot.Box([self.numberOfPointsInXAxis],
-                        ot.Interval([X1min], [X1max])).generate()
-
-            yy = ot.Box([self.numberOfPointsInXAxis],
-                        ot.Interval([X2min], [X2max])).generate()
-
-            xy = ot.Box([self.numberOfPointsInXAxis, self.numberOfPointsInXAxis],
-                        ot.Interval([X1min, X2min], [X1max, X2max])).generate()
-
-            data = self.distribution.computePDF(xy)
+        If :attr:`drawData`, the whole sample is drawn. Otherwise, depending on
+        :attr:`drawOutliers` it will either show the outliers or the inliers
+        only.
 
-            # Label using percentage instead of probability
-            n_contours = len(self.contoursAlpha)
-            labels = ["%.0f %%" % (self.contoursAlpha[i] * 100)
-                      for i in range(n_contours)]
-
-            contour = ot.Contour(xx, yy, data, self.pvalues,
-                                 ot.Description(labels))
-            contour.setColor('black')
-        else:
-            raise NotImplemented('Contour in 2D is not yet supported.')
-
-        return contour
+        :param bool drawData: Plot inliers and outliers.
+        :param bool drawOutliers: Whether to draw inliers or outliers.
+        :returns: figure, axes and OpenTURNS Graph object.
+        :rtypes: Matplotlib figure instances, Matplotlib AxesSubplot instances,
+          :class:`openturns.Graph`
+        """
+        plabels = self.sample.getDescription()
+
+        # Bivariate space
+        fig, sub_ax = plt.subplots()
```

Comment:
My suggestion in drawPairsWithPDF is to replace this line with:

    fig = plt.figure(figsize=(10, 10))

Then the subplots are added within a if condition:

            if i <= j:  # lower triangle
                ax = fig.add_subplot(dim, dim, k)
                graph = ot.Graph('', '', '', True, 'topright')